### ISSUE TITLE
Ethernet/IPv4/TCP net_send error

### ISSUE THREAD

**Issue by nashif** [zephyriot/zep-jira14#382]
**_Reported by Flavio Santes:_**

 net_send function returns success even when no communication can be established.
Workaround: reset the board until net_send works.

1. Open the net_test01.tar.gz file attached to this report, it contains:
      a. Source code used to reproduce the issue.
      b. Network debugging information: galileo_ok.pcapng.gz and galileo_error.pcapng.gz Wireshark files.

(Imported from Jira ZEP-428)

**Comment by nashif** [comment]
**_by Flavio Santes:_**

* [net_test01.tar.gz](https://s3.amazonaws.com/logs.zephyrproject.org/jira/GH-382/net_test01.tar.gz) (size: 3882)

**Comment by nashif** [comment]
**_by Anas Nashif:_**

updates?

**Comment by nashif** [comment]
**_by Jaakko Hannikainen:_**

Fixed in https://gerrit.zephyrproject.org/r/#/c/2777/

**Comment by nashif** [comment]
**_by Flavio Santes:_**

It seems that the RST/ACK response is still not handled in the right way (comment also posted in gerrit #2777).

**Comment by nashif** [comment]
**_by Flavio Santes:_**

This issue is still open, updates?

**Comment by nashif** [comment]
**_by Jaakko Hannikainen:_**

The described problem is actually hard to solve on the current IP stack (the ip stack gets the error status only after the 'ok' is given to the application), and I recommend closing this as wontfix as properly fixing this would cause larger scale refactoring in the to-be-deprecated IP stack, or alternatively would cause very weird workarounds in the application layer.

**Comment by nashif** [comment]
**_by Andrei Laperie:_**

We wontfix this for current stack and solve for an upcoming one.

**Comment by nashif** [comment]
**_by Mark Linkmeyer:_**

_Flavio Santes_ , are you okay with not fixing this bug in the current stack?
_Juan Manuel Cruz Alcaraz_

**Comment by nashif** [comment]
**_by Flavio Santes:_**

_Mark Linkmeyer_   share this decision with the QA team

**Comment by nashif** [comment]
**_by Mark Linkmeyer:_**

_Flavio Santes_ , Zephyr QA is aware of the decision, but in the Zephyr Bug Grooming Session meeting this morning (run by _Sharron LIU_  of Zephyr QA), the team asked if you're okay with not fixing this bug in the current stack since you were the reporter of the bug.  If you're okay with it, please Close the bug.  If you're not okay with it, please explain why in the comments so we can proceed appropriately.  Thx.

**Comment by nashif** [comment]
**_by Flavio Santes:_**

_Mark Linkmeyer_  I guess we can close it.

**Comment by nashif** [comment]
**_by Mark Linkmeyer:_**

Reopened since resolved as Won't Do.

**Comment by nashif** [comment]
**_by Mark Linkmeyer:_**

Closing per agreement from Reporter.

**Comment by nashif** [comment]
**_by Sharron LIU:_**

Reopen according to discussion in the "Bug Grooming Meeting".
GH-600 should root cause to this.

**Comment by nashif** [comment]
**_by Mark Linkmeyer:_**

_Flavio Santes_ , try to replicate this problem on different board/hardware (recommended Arduino 101 with SPI Ethernet driver) and report results.

**Comment by nashif** [comment]
**_by Flavio Santes:_**

The issue is hardware agnostic. Read Jaakko's comment:

---------------
*The described problem is actually hard to solve on the current IP stack (the ip stack gets the error status only after the 'ok' is given to the application), and I recommend closing this as wontfix as properly fixing this would cause larger scale refactoring in the to-be-deprecated IP stack, or alternatively would cause very weird workarounds in the application layer.*

---------------

So, I kindly ask you to reassign this issue to the IP stack owners.

**Comment by nashif** [comment]
**_by Mark Linkmeyer:_**

_Inaky Perez-Gonzalez_ , in today's execution meeting I believe you were one of the people who suggested having Flavio try reproduce the issue on different board/hardware.  Please see his response above.  What needs to happen next on this?

**Comment by nashif** [comment]
**_by Mark Linkmeyer:_**

Hi _Kuo-Lang Tseng_  and _Sharron LIU_ , with Inaky out on vacation now I won't get a response from him in time per my last comment.  So, what do you two suggest be the next steps on this?  Who owns doing what next on this?

**Comment by nashif** [comment]
**_by Kuo-Lang Tseng:_**

_Mark Linkmeyer_  _Sharron LIU_  Based on earlier comments from Andrei and Jaakko, this issue won't be fixed in current IP stack. So with that, I would suggest we should close this ticket. But would ask Flavio to help document the impact of this issue in release note.

**Comment by nashif** [comment]
**_by Flavio Santes:_**

We will update the app's documentation to reflect the reported behavior.

**Comment by nashif** [comment]
**_by Mark Linkmeyer:_**

_Flavio Santes_  and _Kuo-Lang Tseng_ , does it seem appropriate then that we leave this bug open until the app's documentation is updated to reflect the reported behavior -- which essentially then closes this reported bug?  It does to me.  ;-)  We need some way to track that this doc update gets done.  How long will it take to update the doc and by when will it be done?

**Comment by nashif** [comment]
**_by Flavio Santes:_**

Hello _Mark Linkmeyer_ , documentation is being updated. The patch will be submitted (and I hope merged) before the new RC is tagged.

**Comment by nashif** [comment]
**_by Mark Linkmeyer:_**

_Flavio Santes_ , cool!  Then let's leave this in the Reopened state and once you have the documentation updated, go ahead and move it to Closed.

**Comment by nashif** [comment]
**_by Kuo-Lang Tseng:_**

_Mark Linkmeyer_  Makes sense.

**Comment by nashif** [comment]
**_by Flavio Santes:_**

Documentation was updated: https://gerrit.zephyrproject.org/r/#/c/4194/

**Comment by nashif** [comment]
**_by Mark Linkmeyer:_**

Correcting the priority field

**Comment by nashif** [comment]
Blocks GH-371

**Comment by nashif** [comment]
Blocks GH-370

**Comment by nashif** [comment]
Blocks GH-416
